### PR TITLE
Improve GPIO

### PR DESCRIPTION
**Pull request by alexkalmuk**
Rework GPIO general implementation. Tested with STM32F4 and arm/overo

Several additional improvements:
* Improve stm32 cube gpio driver
* Make pin command do not depend on stm32

https://github.com/embox/embox/wiki/GPIO

### CODE REVIEW

File: platform/stm32vl/tests/blinking_led/blinking_led.c [hunk 1]
```diff
@@ -16,16 +16,16 @@ EMBOX_TEST_SUITE("stm32vl blinking_led");
 #define LED_GREEN (1 << 9) /* port C, pin 9 */
 
 static inline void led1_level(unsigned int level) {
-	gpio_set_level(GPIO_C, LED_BLUE, level);
+	gpio_set(GPIO_PORT_CC, LED_BLUE, level);
```
Comment (Deryugin) [inline platform/stm32vl/tests/blinking_led/blinking_led.c]: Typo?

File: src/cmds/hardware/pin/pin.c [hunk 2]
```diff
@@ -144,63 +65,38 @@ int main(int argc, char **argv) {
...
 	}
-	opt++;
-	level = extract_level(opt, argc, argv);
-	apply(gpio, pin, level);
+	pin = (1 << arg);
+
+	if (!strcmp("toggle", argv[opt])) {
+		gpio_setup_mode(gpio, pin, GPIO_MODE_OUTPUT);
+		gpio_toggle(gpio, pin);
+	} else {
```
Comment (Deryugin) [inline src/cmds/hardware/pin/pin.c]: May be `} else { if() ... }` -> `} else if () {}` ?

File: platform/stm32vl/tests/blinking_led/blinking_led.c (hunk 1 above)
Comment (alexkalmuk) [reply platform/stm32vl/tests/blinking_led/blinking_led.c]: Exactly. Thanks, fixed.

File: src/cmds/hardware/pin/pin.c (hunk 2 above)
Comment (alexkalmuk) [reply src/cmds/hardware/pin/pin.c]: Thank you, it looks better. Fixed.

**Pull request merged**